### PR TITLE
Do not suggest things named underscore

### LINKED ISSUE #68719: Invalid "a constant with a similar name exists: `_`" suggestion when a `const _` is in scope
For example ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=const%20_%3A%20()%20%3D%20()%3B%0A%0Afn%20main()%20%7B%0A%20%20%20%20a%0A%7D)):

```rust
const _: () = ();

fn main() {
    a
}
```

Errors with:

```
error[E0425]: cannot find value `a` in this scope
 --> src/main.rs:4:5
  |
1 | const _: () = ();
  | ----------------- similarly named constant `_` defined here
...
4 |     a
  |     ^ help: a constant with a similar name exists: `_`
```

The same also happens with `use Trait as _` ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=pub%20use%20Iterator%20as%20_%3B%0A%0Afn%20foo%3CT%3A%20A%3E(x%3A%20T)%20%7B%7D)):

```rust
pub use Iterator as _;

fn foo<T: A>(x: T) {}
```

```
error[E0405]: cannot find trait `A` in this scope
    --> src/lib.rs:3:11
     |
3    |   fn foo<T: A>(x: T) {}
     |             ^ help: a trait with a similar name exists: `_`
```

The diagnostics shouldn't suggest things named underscore.

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #68719

r? @estebank

### CODE REVIEW

File: src/librustc_resolve/diagnostics.rs [hunk 1]
```diff
@@ -769,6 +769,11 @@ impl<'a> Resolver<'a> {
         span: Span,
     ) -> bool {
         if let Some(suggestion) = suggestion {
+            // We shouldn't suggest underscore.
+            if suggestion.candidate.as_str() == "_" {
```
Comment (Centril) [inline src/librustc_resolve/diagnostics.rs]: You could compare against [`kw::Underscore`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/symbol/kw/constant.Underscore.html) instead, i.e. `if suggestion.candidate == kw::Underscore  { ... }`.

Comment (matthewjasper) [inline src/librustc_resolve/diagnostics.rs]:
```suggestion
            if suggestion.candidate == kw::Underscore {
```

**Comment by JohnTitor** [comment]
Thanks, both! Applied suggestion.

**Pull request merged**